include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -0,0 +1,125 @@
+#ifndef XSPARSE_CO_ITERATION_HPP
+#define XSPARSE_CO_ITERATION_HPP
+
+#include <vector>
+#include <tuple>
+
+namespace xsparse::level_capabilities
+{
+    template <class IK, class PK, class Levels, class IterHelpers>
+    class Coiterate;
+
+    template <class IK, class PK, class... Levels, class... IterHelpers>
+    class Coiterate<IK, PK, std::tuple<Levels...>, std::tuple<IterHelpers...>>
```

Comment:
```suggestion
    class Coiterate<IK, PK, std::tuple<Levels&...>>
```

You can get `iter_helper` from inside the levels.

Reply:
Additionally we only need a reference.

Reply:
But I'm not sure how we are going to get the required IKs and PKs for the level since `iter_helper` takes in `(std::make_tuple(previous IKs...), previous P` as arguments. This might differ for different levels right?